jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp

```diff
@@ -951,6 +950,9 @@ ecma_builtin_array_prototype_object_index_of (ecma_value_t this_arg, /**< this a
     /* 5. */
     ECMA_OP_TO_NUMBER_TRY_CATCH (arg_from_idx, arg2, ret_value);
 
+    ecma_number_t *num_p = ecma_alloc_number ();
+    *num_p = ecma_int32_to_number (-1);
```

Comment:
The lines could be moved to `if` block at line 991, and here we can define variable of `ecma_number_t`, not pointer.
In the case, call to `ecma_dealloc_number` at line 995 could be removed.